### ISSUE TITLE
Make nullable fields on input types optional

### ISSUE THREAD

**Issue by waterfoul** [relay-tools/relay-compiler-language-typescript#46]
When you are sending an input type undefined values are treated the same as nulls, It would be nice to not need to supply these when usng the variables types

**Comment by alloy** [comment]
AFAIK this logic has been ported directly from the Flow type generator. I hear what you’re saying, but I’d prefer that this would be discussed upstream first and see if the Relay team had a specific reason in mind for designing it this way. Could you please file a ticket in the Relay repo with an example of what you’d like to see changed for discussion and link it here?

**Comment by waterfoul** [comment]
I just ran the alloy's 1.5.0-plugin.5 build of the compiler without the language flag and the flow types are using [mabye types](https://flow.org/en/docs/types/maybe/) and [optional properties](https://flow.org/en/docs/types/primitives/#optional-object-properties-) which would be equivalent to a nullable, optional property in typescript

This was copied directly from the output
```
export type SqlString = {
  eq?: ?string,
  ne?: ?string,
  like?: ?string,
  notLike?: ?string,
};
```

**Comment by alloy** [comment]
Aha, so we did indeed not port that fully. Good catch and thanks for the details!

**Comment by ckknight** [comment]
Fixed by #47

**Comment by alloy** [comment]
Finally merged.
